scripts/minimum_deps

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+"""
+Generate a requirements-min.txt file based on install_requres
+"""
+import pkg_resources
+import urllib.request
+import json
+import warnings
+
+
+def get_minimum_version(requirement):
+    """Return minimum version available on PyPi for a given version specification"""
+    content = urllib.request.urlopen(f'https://pypi.python.org/pypi/{requirement.name}/json').read()
+    versions = sorted([pkg_resources.parse_version(v) for v in json.loads(content)["releases"].keys()])
+    # print(requirement)
+    # print(versions)
+    for version in versions:
+        if version in requirement:
+            return version
+    # If there was no version specifier in setup.py, return the 2nd lowest
```

Comment:
I don't think this will actually happen if there was no version specifier -- the `pkg_resources.Requirement` object returns True for all versions if the specifier is missing.  One situation that will cause this is when setup.py contains a requirement for which no version exists.  Seems like that should result in an exception?

Reply:
Ah, right.  It looks it grabs the first version it comes across if there's no version set in `setup.py` requirements.  Hmm.

That said, the purpose of the script is to maintain minimum versions.  So now we have minimum versions for all packages.

I should fix this though.

Reply:
I've clarified comments and added some warning messages.  The script can be used as is I think to iteratively find a solution for the lowest useable version of a package dependency (which is how I used it).  Once the lowest versions are set in `install_requires`, then it should never hit that logic in the code.

I.e. instead of raising an error, I want it to return a possibly not-working version of the dependency so that it can be tested and shown to be too low a version.  This will force maintainers to update that package in `install_requires` to one that works.